### DISCUSSION TITLE
gtk3-1:3.24.42 update breaks monitors assignment

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#330] [General]
Today's gtk3-1:3.24.42-1 update spoiled the `monitor.get_geometry()` function: it returns x = 0 and y = 0 for all monitors. We no longer have a way to assign panels to monitors properly. This way nwg-panel will only appear on the monitor with x = 0, y = 0 coordinates.

**If you use a single monitor**, you should not be affected, as it's always at 0,0. In case I'm wrong: open your panel config and set the `"output"` value empty for each panel you defined, by replacing e.g.: `"output": "eDP-1"` with `"output": ""`.

**If you use multiple monitors**: I can't think of a workaround at the moment. Let's hope it's going to be fixed any time soon. For now you need to downgrade the gtk3 package to 3.24.41.

The bug also affects other nwg-* stuff. Both nwg-dock and nwg-dock-hyprland won't respect the `-o` (output name) argument. Also nwg-displays stopped showing popups on displays of coordinates other than 0,0.

**Comment by philuser-tech** [comment]
Yes, unfortunately I've noticed that. 

 I have another problem with the sleep mode, which freezes, no more actions possible, but my mouse remains active, and I still have access to my other tty terminals. It seems to me that the swayidle/swaylock couple is the cause!!, because this happens as soon as my workstation engages the sleep mode, my only recourse at the moment ... A reboot! 

Is there a workaround for this?

**Comment by philuser-tech** [comment]
In the screen saver/lock settings interface, in the first lock tools field, it seems that you should be able to choose your application, but there's no option other than `swaylock`. I would have liked to have been able to test `hyrplock`.

**Reply by nwg-piotr** [reply]
Yes, I removed gtklock since it stopped working on Hyprland. Will add hyprlock sooner or later.

**Comment by nwg-piotr** [comment]
Back on topic: nwg-panel 0.9.32 (just [released](https://github.com/nwg-piotr/nwg-panel/releases/tag/v0.9.32)) should work well on gtk3-1:3.24.42.

**Reply by philuser-tech** [reply]
Always so responsive and perfect, it's top thank you.

However, I've had a recurring problem for quite some time now: after each validation of the nwg-shell-config panel, it updates the `~/.config/hyrp/includes.conf` file, and then I get an error `..at line72. cannot parse “1.0” as an int. . I had to write a small shell script that I run after each validation of nwg-shell-config to rectify this line 72 from “scroll_button = 1.0” to “scroll_button = 1”. if this could be corrected in the source it would be great!

**Reply by nwg-piotr** [reply]
> I get an error `..at line72. cannot parse “1.0” as an int.

What throws this error message? Hyprland? But the `scroll_factor` value is float, not integer. See: https://wiki.hyprland.org/Configuring/Variables

**Reply by philuser-tech** [reply]
When I click on “Apply” in “nwg-shell-config” to validate the panel screen, I immediately see this famous red banner, and as soon as I rectify this famous line 72, the error disappears! 

![img-wg-shell-config](https://firebasestorage.googleapis.com/v0/b/firescript-577a2.appspot.com/o/imgs%2Fapp%2Fphiluser%2FM3xy7IBpnH.png?alt=media&token=08d8f45e-e156-4568-88d9-307ab479f7cc)

**Reply by philuser-tech** [reply]
If you need any other files or logs, just ask me

**Reply by nwg-piotr** [reply]
Which hyprland version are you on?

**Reply by philuser-tech** [reply]
my hyprland version is: `extra/hyprland 0.40.0-1 (46.4 MiB 53.5 MiB)`

**Reply by nwg-piotr** [reply]
I have no idea what's happening. It doesn't happen on any of my machines, that run the same Hyprland version.

**Reply by philuser-tech** [reply]
Thanks for the effort, it's not very disturbing, I don't change my configurations frequently and it's easy to resolve via my little shell script. Perhaps you could give me a lead or I could investigate?

**Reply by nwg-piotr** [reply]
Well, could you show me your includes file, just in case?

**Reply by philuser-tech** [reply]
here it is `~/.config/hypr/includes.conf` :
```
# This content was generated by nwg-shell-config. Do not modify it manually.

# VARIABLES
$term = foot
$browser = MOZ_ENABLE_WAYLAND=1 firedragon
$filemanager = thunar
$editor = mousepad
$launcher = nwg-drawer
$exit = $launcher

# AUTOSTART
exec-once = rm /tmp/nwg-shell-check-update.lock
exec-once = wlsunset -t 4500 -T 6500 -l 48.8788 -L 2.3223 -g 1.0
exec-once = swaync -c /home/philuser/.config/swaync/hyprland.json -s /home/philuser/.config/swaync/hyprland-0.css
exec-once = nm-applet --indicator
exec-once = wl-paste --type text --watch cliphist store
exec-once = wl-paste --type image --watch cliphist store
exec = nwg-drawer -r -c 6 -is 64 -fscol 2 -s hyprland-0.css -term foot -ft -o eDP-1 -pbexit 'hyprctl dispatch exit' -pblock 'nwg-lock' -pbpoweroff 'systemctl -i poweroff' -pbreboot 'systemctl reboot' -pbsleep 'systemctl suspend' -pbsize 48
exec = nwg-panel -c hyprland-0 -s hyprland-0.css
exec = nwg-dock-hyprland -d -p bottom -l overlay -a center -i 48 -hd 20 -s hyprland-0.css
exec = nwg-shell-check-updates
exec = swayidle timeout 1200 nwg-lock timeout 18000 'swaymsg "output * dpms off"' resume 'swaymsg "output * dpms on"' 
exec-once = nwg-update-indicator
exec = nwg-screenshot-applet

# GENERAL SETTINGS
general {
    border_size = 1
    no_border_on_floating = false
    gaps_in = 3
    gaps_out = 3
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)
    layout = dwindle
    extend_border_grab_area = true
    hover_icon_on_border = true
}

# DWINDLE LAYOUT 
dwindle {
    pseudotile = false
    force_split = 0
    preserve_split = true
    smart_split = false
    smart_resizing = true
    special_scale_factor = 0.8
    split_width_multiplier = 1.0
    no_gaps_when_only = false
    use_active_for_splits = true
    default_split_ratio = 1.0
}

# MASTER LAYOUT 
master {
    allow_small_split = false
    special_scale_factor = 0.8
    mfact = 0.55
    new_is_master = true
    new_on_top = false
    no_gaps_when_only = false
    orientation = left
    inherit_fullscreen = true
    always_center_master = false
}

# INPUT DEVICES
input {
    kb_layout = fr
    repeat_rate = 25
    repeat_delay = 600
    scroll_method = edge
    scroll_button = 1
    follow_mouse = 1
    mouse_refocus = true
    float_switch_override_focus = 1
    touchpad {
        disable_while_typing = true
        scroll_factor = 1.0
        tap-to-click = true
    }
}

# MISC SETTINGS
misc {
    disable_hyprland_logo = true
    disable_splash_rendering = true
    vrr = 0
    mouse_move_enables_dpms = false
    key_press_enables_dpms = false
    layers_hog_keyboard_focus = true
    focus_on_activate = false
    hide_cursor_on_touch = true
    mouse_move_focuses_monitor = true
}

# KEY BINDINGS
bindr = SUPER, SUPER_L, exec, $launcher
```

**Reply by nwg-piotr** [reply]
Of course, the problem was my reading ability. As I used the default 0 value, I had no such line in my includes.conf. What I was talking about was the `scroll_factor` value, which definitely is float. The line that sets the `scroll_button` value used to use an improper function, most likely in result of copy-paste. Fix on the way.
